akka-stream/src/main/scala/akka/stream/QueueOfferResult.scala

```diff
@@ -28,6 +28,14 @@ object QueueOfferResult {
    */
   case object Enqueued extends QueueOfferResult
 
+  /**
+   * Java API: To validate if the element is successfully enqueued.
+   */
+  def isEnqueued: Boolean = enqueued match {
```

Comment:
This is not right, it will always return true since it is just comparing a singleton/constant to itself.

The method should be on the abstract `QueueOfferResult` class, not the companion object. 

You can update the logic in the method `assertSuccess` in `akka.stream.scaladsl.QueueSourceSpec` to use `isEnqueued` instead of comparing the singleton like it does now to verify that you got this right.

Reply:
shoudl add to `sealed abstract class QueueOfferResult`

Reply:
Got your point @johanandren . Thanks for hint @hepin1989 ,
I have refactored it again, also added a case in `akka.stream.scaladsl.QueueSourceSpec` as well. Please lemme know further if required any change.